guide_react.rst

```diff
@@ -0,0 +1,550 @@
+``props`` to build a variety of modular components, we will add 
+read/write/update/delete functionality to this app. From these four basic 
+functions, one can create large-scale systems with ease.
+
+How to Build a To-Do List and Read a To-Do Item
+***********************************************
+
+First, let’s build the read functionality of the app! By the end of this step, 
+we should have a root ``App`` component that looks like this:
+
+.. code-block:: jsx
+
+    const App = () => {
+        const [todos, setTodos] = React.useState([
+            { text: "Learn about React" },
+            { text: "Meet friend for lunch" },
+            { text: "Build a really cool todo app" }
+        ]);
+
+        return (
+            <div className="app">
+            <div className="todo-list">
+                {todos.map((todo, index) => (
+                <Todo
+                    key={index}
+                    index={index}
+                    todo={todo}
+                />
+                ))}
+            </div>
+            </div>
+        );
+    }
+
+Here, we can observe that there are two very important steps to building a 
+component: 1) setting its initial state and 2) returning the element to be 
+rendered.
+
+Let’s take a look at #1. In React, state is just an object that every 
+component uses to store information about itself. In our example, we create 
+a state variable called ``todos`` that holds an array of text items 
+corresponding to our to-do list. It is accompanied by a function called 
+``setTodos`` that we can call at any point within the ``App`` component to change 
+the list. For example, we can add a new to-do item to the end of the list by 
+using the following call:
+
+.. code-block:: javascript
+
+    setTodos([...todos, { text: "Finish the React workshop" }])
+
+When we write ``const [todos, setTodos] = React.useState(/* ... */);``, all 
+we’re really doing is initializing a state variable along with its 
+corresponding setter so that we can control and change the state at any stage 
+of the functional component’s lifecycle. This is why it’s called a Hook; we’re 
+prying into the component’s data and changing it directly from within! React 
+Hooks drastically simplify the dynamic rendering process so that we can edit 
+the content of webpages in real time; no jQuery needed.
+
+.. note::
+
+    Since state is local to each component, it is recommended that you only get 
+    and set a component’s state inside its declaration.
+
+Now, let’s take a look at #2. For any React component, we will always return 
+one HTML element. In this case, we are returning a ``<div>`` container that holds 
+an array of ``<Todo>`` components (``map`` is the method that creates this array). 
+Since ``<Todo>`` isn’t a native HTML element, we have to create this element 
+ourselves by introducing another functional component called ``Todo``, which 
+will return the text for its corresponding to-do item:
```

Comment:
it's unclear where this new component is being created -- same file or different one?